File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day02BathroomSecurityTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day02BathroomSecurityTest  extends FlatSpec with Matchers {
  "2016 Day 02 - Input" should "calculate answers" in {
    val dp = new Day02BathroomSecurity("/twenty_sixteen/Day02-BathroomSecurity-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day02BathroomSecurity("/twenty_sixteen/Day02-BathroomSecurity-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}

File: python-advent-of-code/tests/twenty_sixteen/day02_bathroom_security_test.py
import unittest

from advent.twenty_sixteen.day02_bathroom_security import Day02BathroomSecurity


class Day02BathroomSecurityTestCase(unittest.TestCase):
    def test_real(self):
        dp = Day02BathroomSecurity("data/twenty_sixteen/Day02-BathroomSecurity-input.txt")
        self.assertEqual("14894", dp.part_1_answer)
        self.assertEqual("26B96", dp.part_2_answer)

    def test_example_1(self):
        dp = Day02BathroomSecurity("data/twenty_sixteen/Day02-BathroomSecurity-example#1.txt")
        self.assertEqual("1985", dp.part_1_answer)
        self.assertEqual("5DB3", dp.part_2_answer)


if __name__ == '__main__':
    unittest.main()
